Source/WebCore/platform/graphics/FontCascadeFonts.h

```diff
@@ -75,6 +75,7 @@ class FontCascadeFonts : public RefCounted<FontCascadeFonts> {
     const WidthCache& widthCache() const { return m_widthCache; }
 
     const Font& primaryFont(FontCascadeDescription&);
+    const Font& cachedPrimaryFont(const FontCascadeDescription&);
```

Comment:
The naming here doesn't really communicate what is the difference between these very similar sounding functions. Both actually return a "cached" font.

Reply:
How about `ensurePrimaryFont` and `primaryFont`?

I could call `primaryFont` something like `getPrimaryFontAndResolveReferences` but I'm leaning a bit against changing `primaryFont` name as the resolution of "font-size-adjust from-font".

Otherwise we could have 2 overloads for the function, one that receives a const object and one that receives a mutable object.

Reply:
I ended up splitting this into another commit as it is not directly related to this bug: https://github.com/WebKit/WebKit/pull/27829